src/hooks/autogen.js

```diff
@@ -0,0 +1,72 @@
+const { BadRequest } = require('@feathersjs/errors');
+
+module.exports = {
+  /**
+   * Autogenerates sessions
+   */
+  genSessions: () => (context) => {
+    const { days, dates, coaches } = context.data;
+
+    const oneDay = 24 * 60 * 60 * 1000;
+    const oneWeek = oneDay * 7;
+    const weekDays = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+
+    const start = new Date(dates.start);
+    const end = new Date(dates.end);
+
+    // If none of the days provided match the start date return error
+    let startDay = -1;
```

Comment:
You are writing this like a c or java program you want to use high level language features. Take a look at  some for arrays